### PR TITLE
nullsafety support

### PR DESCRIPTION
**Pull request by vishna**
`r_flutter` generates now null safe code. If you update to v 0.7.0, your project should use dart 2.12. Let's drive the ecosystem to null safety 🚀

Code generator also should be nullsafe, thus this package should get the nullsafe badge on pub.dev

- update example to dart 2.12
- update platform clients
- add macos/web clients
- generate i18n class correctly for nullsafe
- make tests pass
- update codegen's deps to nullsafe (all of them I believe)
- update codegen's code to be null safe

issue #24

### CODE REVIEW

**Comment by timfreiheit** [comment]
awesome. thank you very much. 
I will prepare a release for this tomorrow

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.7.0 (UNRELEASED/BREAKING CHANGE)
+
+  * r_flutter generates now nullsafe code, if you update to v 7.0.0, your project should use dart 2.12
```
Comment (timfreiheit) [inline CHANGELOG.md:3]: this should be 0.7.0

File: lib/src/model/resources.dart [hunk 2]
```diff
@@ -84,14 +84,14 @@ class CustomAssetType extends AssetType {
...
-  final List<String> placeholders;
-  final String value;
-
-  StringReference({
-    this.name,
-    this.placeholders,
-    this.value,
-  });
-}
+// class StringReference {
```
Comment (timfreiheit) [inline lib/src/model/resources.dart:87]: seams like this class can be removed

**Review by timfreiheit — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (vishna) [reply CHANGELOG.md]: fixed

File: lib/src/model/resources.dart (hunk 2 above)
Comment (vishna) [reply lib/src/model/resources.dart]: fixed

**Comment by vishna** [comment]
@timfreiheit pushed the amends

**Pull request merged**